sqlglot/parser.py

```diff
@@ -4484,6 +4492,46 @@ def _parse_as_command(self, start: Token) -> exp.Command:
         size = len(start.text)
         return exp.Command(this=text[:size], expression=text[size:])
 
+    def _parse_dict_property(self, this: str) -> exp.DictProperty:
+        settings = []
+
+        self._match_l_paren()
+        kind = self._parse_id_var()
+        if not kind:
+            self.raise_error(f"Unexpected token")
+        assert kind
+
+        if self._match(TokenType.L_PAREN):
+            while True:
+                key = self._parse_id_var()
+                value = self._parse_primary()
+
+                if not key and value is None:
+                    break
+                settings.append(self.expression(exp.DictSubProperty, this=key, value=value))
+            self._match(TokenType.R_PAREN)
+
+        self._match_r_paren()
+
+        return self.expression(
+            exp.DictProperty,
+            this=this,
+            kind=kind.this,
+            settings=settings,
+        )
+
+    def _parse_dict_range(self, this: str) -> exp.DictRange:
+        self._match_l_paren()
+        has_min = self._match_text_seq("MIN")
+        min = self._parse_var() or self._parse_primary()
+        if has_min:
+            self._match_text_seq("MAX")
+            max = self._parse_var() or self._parse_primary()
+        else:
+            max = min
```

Comment:
why do you need to set max to min here? why not just leave it None?

Reply:
Not correct from the logical standpoint. As CH uses `MIN` and `MAX` as a range `[min, max]` and if there is only one number it's `[number, number]` range internally.
For example if you programmatically change `(MIN 10 MAX 20)` with `MAX = 10` it should produce `(10)`

Reply:
Another idea is to use `(MIN n MAX n)` as a canonical representation of `(n)`

Reply:
yea i'm fine with (min n max n) for n

Reply:
should we remove the str comparison and do

max = max or min

Reply:
Ooof, I was wrong, checked the whole code of how CH builds dictionaries and it's different from what I remembered.
Lifetime has `min_sec = 0` and `max_sec = 0` by default. So, `(n) == (MIN 0 MAX n)`!
Will canonicalize like that!
See [here](https://github.com/ClickHouse/ClickHouse/blob/9f80900d6f587383780d2a40f8173093dce68a5a/src/Parsers/ParserDictionary.cpp#L36)